hase/record/__init__.py

```diff
@@ -82,6 +83,47 @@ def received_coredump(signum: int, frame_type: FrameType) -> None:
 
         return Recording(coredump, trace, exit_code, rusage_result)
 
+def record_process_pid(
+    pid: int,
+    record_paths: "RecordPaths",
+) -> Recording:
+    handler = coredumps.Handler(
+        str(record_paths.coredump),
+        str(record_paths.fifo),
+        str(record_paths.manifest),
+        log_path=str(record_paths.log_path.joinpath("coredump.log")),
+    )
+
+    # work around missing nonlocal keyword in python2 with a list
+    got_coredump = [False]
+
+    def received_coredump(signum: int, frame_type: FrameType) -> None:
+        got_coredump[0] = True
+
+    with IncreasePerfBuffer(100 * 1024), Perf(
+        pid
+    ) as perf, handler as _coredump, SignalHandler(SIGUSR2, received_coredump):
+        write_pid_file(record_paths.pid_file)
+
+        rusage_result = None
+        while True:
+            time.sleep(1)
+            status, output = subprocess.getstatusoutput(f'cat /proc/{pid}/stat')
```

Comment:
I would use `stat` here it is not correctly measured. rusage can be also omitted.